### PR TITLE
Handle null coalescing in the finally block of a constructor.

### PR DESCRIPTION
**Pull request by 333fred**
Fixes https://github.com/dotnet/roslyn/issues/41273.

### CODE REVIEW

File: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs [hunk 1]
```diff
@@ -108753,6 +108752,33 @@ void M1(int? i)
...
+public class C {
+    string x;
+    
+    public C()
+    {
+        try
+        {
+            x = """";
+        }
+        catch (Exception)
```
Comment (gafter) [inline src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: Please add a test like
``` csharp
try {
} finally {
    x ??= "";
}
```

**Comment by 333fred** [comment]
@dotnet/roslyn-compiler for a second review.

**Comment by AlekseyTs** [comment]
@333fred Please make sure the commit comment makes it clear that this change is about flow analysis.

File: src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs [hunk 2]
```diff
@@ -3043,6 +3049,14 @@ public override BoundNode VisitReadOnlySpanFromArray(BoundReadOnlySpanFromArray
             }
         }
 
+        /// <summary>
+        /// This visitor represents just the non-assignment part of the null coalescing assignment
+        /// operator (when the left operand is non-null).
+        /// </summary>
+        protected virtual void VisitUnassignedLeftOfNullCoalescingAssignment(BoundNullCoalescingAssignmentOperator node)
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs]: >VisitUnassignedLeftOfNullCoalescingAssignment [](start = 31, length = 45)

The name of this method feels very confusing to me. It is not meant to visit anything, especially not meant to visit Left. AdjustTheStateForNonNullCase?

**Pull request merged**